src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -250,4 +250,12 @@ object TezosDatabaseOperations extends LazyLogging {
   def doBlocksExist(): DBIO[Boolean] =
     Tables.Blocks.exists.result
 
+  /** count number of the elements in the table */
+  def count(table: TableQuery[_]): DBIO[Int] =
+    table.length.result
+
+  // Slick did not let me to this any other way
+  /** count number of the distinct elements in the table */
+  def countDistinct(table: String, column: String)(implicit ec: ExecutionContext): DBIO[Int] =
+    sql"""SELECT COUNT(DISTINCT #$column) FROM #$table""".as[Int].map(_.head)
```

Comment:
I think it does, but please make sure that the `sql` interpolator does safe-encoding of the passed in args

I don't want someone to ask for table "table; DELETE * from DB" 😁

Reply:
unfortunately it does not(tested locally😄), but int his case this method is used with the only predefined earlier values by slick code gen `TezosDb.countDistinct(table:TableQuery[_].baseTableRow.tableName, col:FieldSymbol.name)`
The best I can do is to put the big warning comment in here, as slick didn't want me to handle this other way than creating custom sql.

Reply:
This, or you may find what's the best way to sanitize sql-injected values (I suspect it might suffice to wrap in quotes?)

I thought too that _now_ it gets called only by safe code, but in the future who knows? We might  need to reuse that code from other callers and forget that it's unsafe (so at least the documented warning might turn out necessary)

Reply:
I think this is the best I could do, as wrapping with quotes does not work and added comment with capital letters.

Reply:
Let x be some defined table, like Table.Accounts, for example. Wouldn't x.distinct.length do the same thing? Maybe there's confusion because slick's countDistinct function is deprecated?

Reply:
@ba205 there is function combination `distinctOn(<column>).length` in slick but I couldn't pass the column as an argument to this method so I had to do the plain sql - writing generic code in slick is tricky